docs/source/conf.py

```diff
@@ -288,6 +288,14 @@
 # documentation.
 #
 
+switcher_version = version
+if ".dev" in version:
+    switcher_version = "dev/"
+else:
+    # If we are not building dev version of the docs, we are building
+    # docs for the stable version
+    switcher_version = ""
```

Comment:
I just realize a potential problem with this .. We initially build the documentation as the stable docs for the release, but then afterwards we copy those those docs into a subdirectory to keep them as older docs, at which point they are not the "stable" docs anymore ...

Reply:
Uh, I have been thinking about this today and am getting very confused =D

Reply:
I think we will need to check for the numbered version of the docs here (13.0.0, 14.0.0, etc) and set the `switcher_version` to the number extracted from the version string (if not dev) so that once the docs get copied into the subdirectory will be correctly and equal to the version in the `versions.json`.

Then all the docs versions (old and dev), except stable, should have the correct text in the version switcher button. For the stable version - should we open a PR upstream to change the default text? 😊

Reply:
Actually, can we just include both entries in the json file? Both one with "" for the old docs, and one with the actual version for the stable docs, so something like:

```
    {
        "name": "13.0 (stable)",
        "version": "13.0.0/",
        "url": "https://arrow.apache.org/docs/",
        "preferred": true
    },
    {
        "name": "13.0 (stable)",
        "version": "",
        "url": "https://arrow.apache.org/docs/",
    },
```

?

Reply:
Although that would then give a duplicated entry in the dropdown, so also not ideal I suppose.

Another option would be to patch the doc sources when moving them from `/` to `/xx.x` when doing a release. It touches every file, but it's also a quite simple patch. A simple replacement of `DOCUMENTATION_OPTIONS.theme_switcher_version_match = '';` with the correct version.

Reply:
I am in favour of the second approach (patch the doc sources when moving) 👍 
So this PR can be merged as is, the patch can be added as a follow up after the release, right?

Reply:
All in all, I would maybe just merge this PR as is, and then deal with the remaining issue discussed above to when we actually did a release, then it will be easier to experiment with this